contracts/upgradeable_contracts/ChaiConnector.sol

```diff
@@ -270,6 +270,9 @@ contract ChaiConnector is Ownable, ERC20Bridge {
             uint256 initialDaiBalance = daiBalance();
             chaiToken().draw(address(this), amount);
             uint256 redeemed = daiBalance() - initialDaiBalance;
+
+            require(redeemed >= amount);
```

Comment:
Don't you need a similar check in the case if `chaiToken().draw(address(this), invested)` is invoked?